### PR TITLE
testing/telegraf: new aport

### PR DESCRIPTION
**Pull request by Fusl**

### CODE REVIEW

**Comment by kaniini** [comment]
Does this work under `abuild rootbld`?  I believe glide still requires network access, so we may need to  have options="!net" here.

**Comment by Fusl** [comment]
Yes, it requires network access. Adding !net now and re-committing.

**Comment by Fusl** [comment]
Updated to [latest version `1.6.1`](https://github.com/influxdata/telegraf/releases/tag/1.6.1). @kaniini any chance on this getting finally merged?

**Comment by andypost** [comment]
Latest version is 1.7.3 and #4383 does  ot use glide

**Comment by Fusl** [comment]
Updated to [latest version `1.7.3`](https://github.com/influxdata/telegraf/releases/tag/1.7.3).

File: testing/telegraf/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+pkgusers="telegraf"
+pkggroups="telegraf"
+install="$pkgname.pre-install"
+options="!strip !net"
+source="$pkgname-$pkgver.tar.gz::https://github.com/influxdata/$pkgname/archive/$pkgver.tar.gz
+	telegraf-makefile-ldflags.patch
+	telegraf.initd
+	telegraf.logrotate
+	glide.yaml
+	glide.lock
```
Comment (ncopa) [inline testing/telegraf/APKBUILD]: Good that you use glide and a lockfile :+1:

File: testing/telegraf/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	mkdir -p "$_godir"/bin "$_godir"/src/github.com/"$_vendor" "$builddir/.glide" "$builddir/.glidetmp"
+	ln -sf "$builddir" "$_godir"/src/github.com/$_vendor/$pkgname
+	cp "$srcdir"/glide.yaml "$srcdir"/glide.lock "$_godir"/src/github.com/$_vendor/$pkgname
+}
+
+build() {
+	cd "$_godir"/src/github.com/$_vendor/$pkgname
+	export GOPATH="$_godir"
+	export PATH="$PATH:$_godir"
+	GLIDE_HOME="$builddir/.glide" GLIDE_TMP="$builddir/.glidetmp" glide install
```
Comment (ncopa) [inline testing/telegraf/APKBUILD]: good that you set `GLIDE_HOME` so it does not pollute users $HOME, and that you set `GLIDE_TMP` :+1:

File: testing/telegraf/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	make PREFIX=/usr VERSION=$pkgver BRANCH= COMMIT= telegraf
+}
+
+check() {
+	cd "$_godir"/src/github.com/$_vendor/$pkgname
+	make test
+}
+
+package() {
+	mkdir -p "$pkgdir"/usr/bin "$pkgdir"/etc/$pkgname "$pkgdir"/etc/logrotate.d/$pkgname
```
Comment (ncopa) [inline testing/telegraf/APKBUILD]: Good that you also thought of logrotate :+1:

**Pull request closed without merging**